src/main/app/intercom.ts

```diff
@@ -38,16 +38,17 @@ function handleShowOnboardingScreens(showWelcomeScreen: boolean, showNewServerMo
     log.debug('handleShowOnboardingScreens', {showWelcomeScreen, showNewServerModal, mainWindowIsVisible});
 
     if (showWelcomeScreen) {
-        if (ModalManager.isModalDisplayed()) {
+        const welcomeScreen = ModalManager.modalQueue.find((modal) => modal.key === 'welcomeScreen');
```

Comment:
not major, but it would be good to have `welcomeScreen` as a constant

Reply:
Agreed, we should actually do this across the board for all of the modal names.
Seems like something AI could help me hash out real quick ;)

I'll put it in a follow-up change though.

Reply:
https://mattermost.atlassian.net/browse/MM-63265